projects/sbb-esta/angular-business/chip/src/chip-input/chip-input.component.ts

```diff
@@ -0,0 +1,238 @@
+import { coerceBooleanProperty } from '@angular/cdk/coercion';
+import {
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  EventEmitter,
+  HostBinding,
+  Input,
+  OnInit,
+  Optional,
+  Output,
+  Self
+} from '@angular/core';
+import { NgControl } from '@angular/forms';
+import {
+  AutocompleteComponent,
+  AutocompleteOriginDirective
+} from '@sbb-esta/angular-business/autocomplete';
+import { FormFieldControl } from '@sbb-esta/angular-core/forms';
+import { SBB_OPTION_PARENT_COMPONENT } from '@sbb-esta/angular-public';
+import { Subject } from 'rxjs';
+
+let nextUniqueId = 0;
+
+@Component({
+  selector: 'sbb-chip-input',
+  templateUrl: './chip-input.component.html',
+  styleUrls: ['./chip-input.component.scss'],
+  providers: [{ provide: SBB_OPTION_PARENT_COMPONENT, useExisting: ChipInputComponent }]
+})
+export class ChipInputComponent implements FormFieldControl<any>, OnInit {
+  /** Value of the select control. */
+  @Input()
+  options: string[] = [];
+
+  /** Optional autocomplete Component */
+  @Input('sbbAutocomplete')
+  autocomplete: AutocompleteComponent;
+
+  /** Disables the chip input */
+  @Input()
+  @HostBinding('class.sbb-chip-input-disabled')
+  get disabled() {
+    return this._disabled;
+  }
+  set disabled(value: any) {
+    this._disabled = coerceBooleanProperty(value);
+  }
+  private _disabled = false;
+
+  /** Unique id of the element. */
+  @Input()
+  @HostBinding('attr.id')
+  get id(): string {
+    return this._id;
+  }
+  set id(value: string) {
+    this._id = value || this._uid;
+    this.stateChanges.next();
+  }
+  private _id: string;
+
+  /** Whether the component is required. */
+  @HostBinding('class.sbb-select-required')
+  @Input()
+  get required(): boolean {
+    return this._required;
+  }
+  set required(value: boolean) {
+    this._required = coerceBooleanProperty(value);
+    this.stateChanges.next();
+  }
+
+  @HostBinding('class.sbb-chip-input-active')
+  get isActive() {
+    return !this.disabled && this.focused;
+  }
+
+  /** Value of the chip input control. */
+  get value(): any {
+    return this._value;
+  }
+  set value(newValue: any) {
+    if (newValue !== this._value) {
+      this._value = newValue;
+    }
+  }
+  private _value: any;
```

Comment:
Depends on the implementation, I tend to leave value initialization up to the consumer, but I never worked on building a library, so it might make sense to avoid some easy empty value errors right off the bat in case the consumer "forgets" or fails to initialize this value.

Reply:
Sounds good 👍